public/app/features/plugins/plugin_loader.ts

```diff
@@ -218,3 +228,21 @@ export async function importAppPlugin(meta: PluginMeta): Promise<AppPlugin> {
 
   return plugin;
 }
+
+// exported for testing only
+export async function addLocalesToI18n(pluginId: string, locales: Record<string, string>): Promise<void[]> {
+  const promises = Object.entries(locales).map(async ([lang, path]) => {
+    try {
+      const module = await SystemJS.import(resolveModulePath(path));
+      if (!module.default) {
+        return;
+      }
+
+      getI18next().addResourceBundle(lang, pluginId, module.default, undefined, true);
+    } catch (e) {
+      console.warn('Could not load locale for plugin', pluginId, lang, e);
+    }
+  });
+
+  return Promise.all(promises);
```

Comment:
@joshhunt @ashharrison90 do you think it's worth optimizing this so we only load the current loaded language in Grafana? Only because we do a full page refresh on language change that is. If we were to change language "on the fly" then we might need to rethink this. WDYT?

Reply:
I was going to suggest only loading the current locale or falling back to the default language due to the full page refresh on language change.

Reply:
Fixed 👍